projects/kit/src/lib/masks/number/processors/thousand-separator-postprocessor.ts

```diff
@@ -22,21 +23,28 @@ export function createThousandSeparatorPostprocessor({
         return identity;
     }
 
-    const prefixReg = new RegExp(`^${escapeRegExp(prefix)}${CHAR_MINUS}?`);
-    const postfixReg = new RegExp(`${escapeRegExp(postfix)}$`);
+    const minusReg = new RegExp(`^${CHAR_MINUS}?`);
     const isAllSpaces = (...chars: string[]): boolean => chars.every(x => /\s/.test(x));
 
     return ({value, selection}) => {
-        const [integerPart, decimalPart = ''] = value.split(decimalSeparator);
+        const {
+            cleanValue: cleanValueWithPossibleMinus,
+            extractedPostfix,
+            extractedPrefix,
+        } = extractAffixes(value, {
+            prefix,
+            postfix,
+        });
+
+        const [extractedMinus = ''] = cleanValueWithPossibleMinus.match(minusReg) || [];
+
+        const cleanValue = cleanValueWithPossibleMinus.replace(minusReg, '');
+
+        const [integerPart, decimalPart = ''] = cleanValue.split(decimalSeparator);
```

Comment:
```ts
    const {cleanValue, extractedPostfix, extractedPrefix} = extractAffixes(value, {
        prefix,
        postfix,
    });
    
    // drop this line
    // const [extractedMinus = ''] = cleanValue.match(minusReg) || [];

    const [integerPart, decimalPart = ''] = cleanValue
        .replace(CHAR_MINUS, '')
        .split(decimalSeparator);

    // [...]
    return {
            value:
                extractedPrefix +
                (cleanValue.includes(CHAR_MINUS) ? CHAR_MINUS : '') +
                processedIntegerPart +
                (cleanValue.includes(decimalSeparator) ? decimalSeparator : '') +
                decimalPart +
                extractedPostfix,
```

(the only possible minus inside `cleanValue` is negative sign => we can use just `includes`)

Reply:
\+ remove `minusReg` constant